SignalServiceKit/src/Protos/Generated/SSKProto.swift

```diff
@@ -0,0 +1,3163 @@
+        switch value {
+        case .unknown: return .unknown
+        case .ciphertext: return .ciphertext
+        case .keyExchange: return .keyExchange
+        case .prekeyBundle: return .prekeyBundle
+        case .receipt: return .receipt
+        }
+    }
+
+    private class func SSKProtoEnvelopeTypeUnwrap(_ value: SSKProtoEnvelopeType) -> SignalServiceProtos_Envelope.TypeEnum {
+        switch value {
+        case .unknown: return .unknown
+        case .ciphertext: return .ciphertext
+        case .keyExchange: return .keyExchange
+        case .prekeyBundle: return .prekeyBundle
+        case .receipt: return .receipt
+        }
+    }
+
+    @objc public let type: SSKProtoEnvelopeType
+    @objc public let source: String?
+    @objc public let sourceDevice: UInt32
+    @objc public let relay: String?
+    @objc public let timestamp: UInt64
+    @objc public let legacyMessage: Data?
+    @objc public let content: Data?
+
+    @objc public init(type: SSKProtoEnvelopeType,
+                      source: String?,
+                      sourceDevice: UInt32,
+                      relay: String?,
+                      timestamp: UInt64,
+                      legacyMessage: Data?,
+                      content: Data?) {
+        self.type = type
+        self.source = source
+        self.sourceDevice = sourceDevice
+        self.relay = relay
+        self.timestamp = timestamp
+        self.legacyMessage = legacyMessage
+        self.content = content
+    }
+
+    @objc
+    public func serializedData() throws -> Data {
+        return try self.asProtobuf.serializedData()
+    }
+
+    @objc public class func parseData(_ serializedData: Data) throws -> SSKProtoEnvelope {
+        let proto = try SignalServiceProtos_Envelope(serializedData: serializedData)
+        return try parseProto(proto)
+    }
+
+    fileprivate class func parseProto(_ proto: SignalServiceProtos_Envelope) throws -> SSKProtoEnvelope {
+        var type: SSKProtoEnvelopeType = .unknown
+        if proto.hasType {
+            type = SSKProtoEnvelopeTypeWrap(proto.type)
+        }
+
+        var source: String? = nil
+        if proto.hasSource {
+            source = proto.source
+        }
+
+        var sourceDevice: UInt32 = 0
+        if proto.hasSourceDevice {
+            sourceDevice = proto.sourceDevice
+        }
+
+        var relay: String? = nil
+        if proto.hasRelay {
+            relay = proto.relay
+        }
+
+        var timestamp: UInt64 = 0
+        if proto.hasTimestamp {
+            timestamp = proto.timestamp
+        }
+
+        var legacyMessage: Data? = nil
+        if proto.hasLegacyMessage {
+            legacyMessage = proto.legacyMessage
+        }
+
+        var content: Data? = nil
+        if proto.hasContent {
+            content = proto.content
+        }
+
+        // MARK: - Begin Validation Logic for SSKProtoEnvelope -
```

Comment:
The generate code leaves boundary markers for custom validation code. Custom code is preserved when updating the generated code.